utopian-io/v2.utopian.io
# Feature/auth-app/UT-203

### Verify that

Implements [UT-203](https://utopian-io.atlassian.net/browse/UT-203)

This is how the auth app works:
* Separate quasar app 
* `yarn dev` runs the auth app on `localhost:8081`
* Redirect the user to `localhost:8081/login?redirectUrl=myurl`
* The login page will show up and after the user connect his GitHub account he will be redirected to `redirectUrl`

### **Test plan**
* Run the app with `yarn dev` on the root folder
* Verify that `@utopian/api` is running on port 5000, `@utopian/client` is running on port 8080 and `@utopian/auth` is running on port 8081

_**Scenario 1** User doesn't have account_

* Go to localhost:8080
* Click on the `Sign in` button in the toolbar
* You will be redirected to `localhost:8081/login`
* Click on the `Sign in with GitHub` button (translation missing)
* You will be redirected to `/users/create`
* Create a new account

**Expected result:** You should be redirected to the main app page with a signed in user. Verify that the `access_token` and the `refresh_token` are set in the store and in the cookies.

_**Scenario 2** User has an account_

* Go to localhost:8080
* Click on the `Sign in` button in the toolbar
* You will be redirected to `localhost:8081/login`
* Click on the `Sign in with GitHub` button (translation missing)

**Expected result:** You should be redirected to the main app page with a signed in user. Verify that the `access_token` and the `refresh_token` are set in the store and in the cookies.
